instance/import.go

```diff
@@ -0,0 +1,187 @@
+							log.ErrorCtx(ctx, errors.WithMessage(err, notFoundErr), logData)
+							return &taskError{errors.New(notFoundErr), http.StatusNotFound}
+						}
+						log.ErrorCtx(ctx, errors.WithMessage(err, "failed to update build hierarchy task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			}
+			if !hasHierarchyImportTask {
+				validationErrs = append(validationErrs, errors.New("bad request - missing hierarchy task"))
+			}
+		}
+
+		if tasks.BuildSearchIndexTasks != nil {
+			hasImportTasks = true
+			var hasSearchIndexImportTask bool
+			for _, task := range tasks.BuildSearchIndexTasks {
+				hasSearchIndexImportTask = true
+				if err := models.ValidateImportTask(task.GenericTaskDetails); err != nil {
+					validationErrs = append(validationErrs, err)
+				} else {
+					if err := s.UpdateBuildSearchTaskState(instanceID, task.DimensionName, task.State); err != nil {
+						if err.Error() == "not found" {
+							notFoundErr := task.DimensionName + " search index import task does not exist"
+							log.ErrorCtx(ctx, errors.WithMessage(err, notFoundErr), logData)
+							return &taskError{errors.New(notFoundErr), http.StatusNotFound}
+						}
+						log.ErrorCtx(ctx, errors.WithMessage(err, "failed to update build hierarchy task state"), logData)
+						return &taskError{err, http.StatusInternalServerError}
+					}
+				}
+			}
+			if !hasSearchIndexImportTask {
+				validationErrs = append(validationErrs, errors.New("bad request - missing search index task"))
+			}
+		}
+
+		if !hasImportTasks {
+			validationErrs = append(validationErrs, errors.New("bad request - request body does not contain any import tasks"))
+		}
+
+		if len(validationErrs) > 0 {
+			for _, err := range validationErrs {
+				log.ErrorCtx(ctx, errors.WithMessage(err, "validation error"), logData)
+			}
+			// todo: add all validation errors to the response
+			return &taskError{validationErrs[0], http.StatusBadRequest}
+		}
+		return nil
+	}()
+
+	if updateErr != nil {
+		if auditErr := s.Auditor.Record(ctx, UpdateImportTasksAction, audit.Unsuccessful, auditParams); auditErr != nil {
+			updateErr = &taskError{errs.ErrInternalServer, http.StatusInternalServerError}
+		}
+		log.ErrorCtx(ctx, errors.WithMessage(updateErr, "updateImportTask endpoint: request unsuccessful"), logData)
+		http.Error(w, updateErr.Error(), updateErr.status)
+		return
+	}
+
+	if auditErr := s.Auditor.Record(ctx, UpdateImportTasksAction, audit.Successful, auditParams); auditErr != nil {
+		return
+	}
+
+	log.InfoCtx(ctx, "updateImportTask endpoint: request successful", logData)
+}
+
+func unmarshalImportTasks(reader io.Reader) (*models.InstanceImportTasks, error) {
+
+	b, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, errs.ErrUnableToReadMessage
+	}
+
+	var tasks models.InstanceImportTasks
+	err = json.Unmarshal(b, &tasks)
+	if err != nil {
+		return nil, errs.ErrUnableToParseJSON
```

Comment:
are we losing the context of _why_ by doing this? iirc json.Unmarshal gives more info about how far it got through parsing the JSON and why it failed?

Reply:
have returned the plain error instead and added a test case to prove the case where extra information is provided